contrib/reformat-code.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/python3
+#
+# Copyright (C) 2017 Dell Inc.
+#
+# SPDX-License-Identifier: LGPL-2.1+
+#
+
+import argparse
+import os
+import sys
+import subprocess
+
+CLANG_FORMATTERS = [
+    "clang-format-13",
+    "clang-format",
+]
+FIXUPS = {"g_autoptr (": "g_autoptr(", "sizeof (": "sizeof(", "g_auto (": "g_auto("}
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="Reformat code to match style")
+    parser.add_argument("files", nargs="*", help="files to reformat")
+    args = parser.parse_args()
+    if len(args.files) == 0:
+        parser.print_help()
+        sys.exit(0)
+    return args
+
+
+def select_clang_version():
+    for formatter in CLANG_FORMATTERS:
+        try:
+            ret = subprocess.check_call([formatter, "--version"])
+            if ret == 0:
+                return formatter
+        except FileNotFoundError:
+            continue
+    return None
+
+
+def reformat_file(formatter, f):
+    print("Reformatting %s using %s" % (f, formatter))
+    ret = subprocess.check_call([formatter, "-i", "-style=file", f])
+    if ret:
+        print("Failed to run formatter")
+        sys.exit(1)
+    lines = None
+    with open(f, "r") as rfd:
+        lines = rfd.readlines()
+    with open(f, "w") as wfd:
+        for line in lines:
+            for fixup in FIXUPS:
+                if fixup in line:
+                    line = line.replace(fixup, FIXUPS[fixup])
+            wfd.write(line)
```

Comment:
is this going to cause the file time to change every time?  I wonder if the file should just be cached and only spit out if a fixup was run.

Reply:
Does this matter? If the file does not change git would not stage it

Reply:
Doesn't it cause mtime changes to open and write it out even if it's identical?

Reply:
As an example: `touch COMMITMENT; git add COMMITMENT` yields no staged changes

Reply:
hmm, but it makes my editor reload the file. can we add a save = True if `if fixup in line` and then only write if that's set?

Reply:
Sadly, the clang-format will introduce the space on every run for the script to remove it after. I'll try to get around this by not having clang-format to replace the file in place

Reply:
Done